zetaclient/cctx_scanner.go

```diff
@@ -0,0 +1,227 @@
+package zetaclient
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"sort"
+
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+
+	"github.com/rs/zerolog"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	clienttypes "github.com/zeta-chain/zetacore/zetaclient/types"
+)
+
+const (
+	RescanBatchSize uint64 = 1000
+)
+
+// CctxScanner scans missed pending cctx and updates their status
+type CctxScanner struct {
+	tssPubkey         string
+	db                *gorm.DB
+	logger            *zerolog.Logger
+	bridge            *ZetaCoreBridge
+	nextNonceToScan   map[int64]uint64                                   // chainID -> next nonce to scan from
+	missedPendingCctx map[int64]map[uint64]*crosschaintypes.CrossChainTx // chainID -> nonce -> missed pending cctx
+}
+
+func NewCctxScanner(bridge *ZetaCoreBridge, dbpath string, memDB bool, tssPubkey string, logger *zerolog.Logger) (*CctxScanner, error) {
+	sc := &CctxScanner{
+		logger:            logger,
+		bridge:            bridge,
+		nextNonceToScan:   make(map[int64]uint64),
+		missedPendingCctx: make(map[int64]map[uint64]*crosschaintypes.CrossChainTx),
+	}
+	err := sc.LoadDB(dbpath, memDB)
+	if err != nil {
+		return nil, err
+	}
+
+	// on bootstrap or tss migration
+	if tssPubkey != sc.tssPubkey {
+		err = sc.Reset(tssPubkey)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return sc, nil
+}
+
+// Scan a new batch of missed pending cctx
```

Comment:
```suggestion
// ScanMissedPendingCctx scans a new batch of missed pending cctx
```

Go convention for comment of functions are: `[FunctionName] does ....`

Same review comment for functions below